lib/salus/scanners/language_version/go_version_scanner.rb

```diff
@@ -0,0 +1,44 @@
+require 'salus/scanners/language_version/base'
+
+module Salus::Scanners::LanguageVersion
+  class GoVersionScanner < Base
+    def self.supported_languages
+      %w[go]
+    end
+
+    private
+
+    def lang_version
+      @lang_version ||= go_project? ? go_version : nil
+    end
+
+    def run_version_scan?
+      go_project?
+    end
+
+    # If we run 'go mod edit -json' for a 'go' project,
+    # it returns the details of go mod in json format.
+    # We can parse the json file to get the 'go' language version used
+    # Following is a sample output for running 'go mod edit -json':
+    # {
+    # "Module": {
+    #   "Path": "example.com/mymodule"
+    # },
+    # "Go": "1.16",
+    #  ...}
+    def go_version
+      shell_return = Dir.chdir(@repository.path_to_repo) do
```

Comment:
The Dir.chdir isn't threadsafe.  That's not a deal breaker now but it would be a fan of pulling out this block of code

https://github.com/coinbase/salus/blob/master/lib/salus/scanners/base.rb#L125-L131

So you could reference here too, that will normalize our system calls to a common block of code, both for standard scanners and your language scanners here.

Reply:
More of note to self for our concurrent PR which will need to change this block.

Reply:
@joshua-ostrom: Is there any thread-safe solution for this? or, Do we have any future work to resolve thread-safety issue in Salus? 🤔

Reply:
Yeah I'm testing my thread safe version as we speak ;-) I added a chdir param to the run_shell.  You're good here it will be easy to refactor to use it once we merge.